packages/driver/src/httpScram.ts

```diff
@@ -24,6 +24,31 @@ function b64ToUtf8(str: string): string {
   return utf8Decoder.decode(decodeB64(str));
 }
 
+/**
+ * Parses SCRAM-SHA-256 authentication parameters from a string.
+ * @param authParamsString a string of comma-separated key-value pairs
+ * @returns an object with `sid` and `data` properties, decoded string or `null` when not present
+ */
+function parseSidAndData(paramsStr: string): {
+  sid: string | null;
+  data: string | null;
+} {
+  const params = new Map(
+    paramsStr.length > 0
+      ? paramsStr
+          .split(",")
+          .map((attr) => attr.split(/=(.+)?/, 2)) // split on first '=' only; nb, `.split("=", 2)` doesn't do that
```

Comment:
It would be good to add a test, if reasonable, to the suite that exercises when this doesn't work especially since this changes the existing behavior. Wouldn't want to regress on this if indeed we need the `[first, ...rest]` kind of behavior here.

Reply:
Thanks for the detailed explanation and investigation. I think what you have here is clearly correct and I'd really rather keep the change you made and harden it by testing it, but I realize this isn't straightforward to test from the outside of this module. This PR conversation can be our documentation about the issue until such a time as we add some tests that actually exercise this logic.

🙏